test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/AttributeRouteTest.cs

```diff
@@ -545,6 +545,118 @@ public void AttributeRoute_ThrowsRouteCreationException_ForConstraintsNotTakingA
             Assert.IsType<RouteCreationException>(exception.InnerException);
         }
 
+        [Fact]
+        public void GetEntries_DoesNotCreateOutboundEntriesForAttributesWithSuppressForLinkGenerationSetToTrue()
+        {
+            // Arrange
+            var actions = new List<ActionDescriptor>()
+            {
+                new ActionDescriptor()
+                {
+                    AttributeRouteInfo = new AttributeRouteInfo
+                    {
+                        Template = "blog/get/{id}",
+                        Name = "BLOG_LINK1",
+                        SuppressForLinkGeneration = true,
+                    },
+                },
+                new ActionDescriptor()
+                {
+                    AttributeRouteInfo = new AttributeRouteInfo
+                    {
+                        Template = "blog/{snake-cased-name}",
+                        Name = "BLOG_INDEX2",
+                    },
+                },
+                new ActionDescriptor()
+                {
+                    AttributeRouteInfo = new AttributeRouteInfo
+                    {
+                        Template = "blog/",
+                        Name = "BLOG_HOME",
+                        SuppressForPathMatching = true,
+                    },
+                },
+            };
+
+            var builder = CreateBuilder();
+            var actionDescriptorProvider = CreateActionDescriptorProvider(actions);
+            var route = CreateRoute(CreateHandler().Object, actionDescriptorProvider.Object);
+
+            // Act
+            route.AddEntries(builder, actionDescriptorProvider.Object.ActionDescriptors);
+
+            // Assert
+            Assert.Collection(
+                builder.OutboundEntries,
+                e =>
+                {
+                    Assert.Equal("BLOG_INDEX2", e.RouteName);
+                    Assert.Equal("blog/{snake-cased-name}", e.RouteTemplate.TemplateText);
+                },
+                e =>
+                {
+                    Assert.Equal("BLOG_HOME", e.RouteName);
+                    Assert.Equal("blog/", e.RouteTemplate.TemplateText);
+                });
+        }
+
+        [Fact]
+        public void GetEntries_DoesNotCreateOutboundEntriesForAttributesWithSuppressForPathMatchingSetToTrue()
```

Comment:
Would like to see a case for inbound with two actions with the same route info - and one of them is suppressed.